File: backend_python/src/todo_api/domain/repository/todo_repository.py
from abc import ABC, abstractmethod

from todo_api.domain.model.todo import Todo
from todo_api.utils.uuid import UUID7


class TodoRepository(ABC):
    @abstractmethod
    def find_all(self) -> list[Todo]:
        pass

    @abstractmethod
    def find_by_id(self, todo_id: UUID7) -> Todo:
        pass

    @abstractmethod
    def save(self, todo: Todo) -> None:
        pass

    @abstractmethod
    def delete(self, todo: Todo) -> None:
        pass

File: backend_java/src/main/java/io/github/ikedadada/backend_java/domain/repository/TodoRepository.java
package io.github.ikedadada.backend_java.domain.repository;

import java.util.ArrayList;
import java.util.UUID;

import io.github.ikedadada.backend_java.domain.model.Todo;

public interface TodoRepository {
    ArrayList<Todo> findAll();

    Todo findById(UUID id) throws RepositoryException.TodoNotFound;

    void save(Todo todo);

    void delete(UUID id);
}
